include/zephyr/sys/util.h

```diff
@@ -421,6 +421,16 @@ static inline bool is_power_of_two(unsigned int x)
 	return IS_POWER_OF_TWO(x);
 }
 
+/**
+ * @brief Is @p p equal to ``NULL``?
```

Comment:
Can we add a comment here explaining that the point of this API is avoid compiler warnings when comparing values against build-time non-NULL values?  Otherwise folks like me are likely to come along and think "This is silly, let's remove it".

Reply:
Given its potential widespread applicability, if we relocate this function to `sys/util.h`, we should consider renaming it to something more distinctive. This precaution helps avoid confusion and prevents the indiscriminate use of this function in every NULL check scenario.

How about:
- `check_null_macro_silence()`
- `is_null_no_warn_macro()`
- `check_null_quietly_in_macro()`

Reply:
I'll add some additional documentation on the purpose of this.

Yes, perhaps renaming it would be a good idea too. How about `is_null_no_warn()`?

Reply:
I have renamed the function to `is_null_no_warn()` and added more documentation. PTAL.